File: app/src/main/java/fr/frogdevelopment/pregnancycalendar/data/ContractionDao.java
package fr.frogdevelopment.pregnancycalendar.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import io.reactivex.Single;

@Dao
public interface ContractionDao {

    @Query("SELECT * FROM contraction")
    LiveData<List<Contraction>> getAll();

    @Insert
    Single<Long> insert(Contraction contraction);

    @Delete
    void delete(Contraction contraction);

    @Query("DELETE FROM contraction")
    void deleteAll();
}

File: app/src/main/java/fr/frogdevelopment/pregnancycalendar/data/ContractionDao.kt
package fr.frogdevelopment.pregnancycalendar.data

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface ContractionDao {

    @Query("SELECT * FROM contraction ORDER BY datetime ASC")
    fun all(): LiveData<List<Contraction>>

    @Insert
    suspend fun insert(contraction: Contraction?)

    @Delete
    suspend fun delete(contraction: Contraction?)

    @Query("DELETE FROM contraction")
    suspend fun deleteAll()
}